apps/dash-web-trader/app.py

```diff
@@ -0,0 +1,1447 @@
+                ],
+            ),
+            # Contents
+            html.Div(
+                id=current_row[0] + "contents",
+                className="row details",
+                children=[
+                    # Button for buy/sell modal
+                    html.Div(
+                        className="button-buy-sell-chart",
+                        children=[
+                            html.Button(
+                                id=current_row[0] + "Buy",
+                                children="Buy/Sell",
+                                n_clicks=0,
+                            )
+                        ],
+                    ),
+                    # Button to display currency pair chart
+                    html.Div(
+                        className="button-buy-sell-chart-right",
+                        children=[
+                            html.Button(
+                                id=current_row[0] + "Button_chart",
+                                children="Chart",
+                                n_clicks=1
+                                if current_row[0] in ["EURUSD", "USDCHF"]
+                                else 0,
+                            )
+                        ],
+                    ),
+                ],
+            ),
+        ]
+    )
+
+
+# color of Bid & Ask rates
+def get_color(a, b):
+    if a == b:
+        return "white"
+    elif a > b:
+        return "#45df7e"
+    else:
+        return "#da5657"
+
+
+# Replace ask_bid row for currency pair with colored values
+def replace_row(currency_pair, index, bid, ask):
+    index = index + 1  # index of new data row
+    new_row = (
+        currency_pair_data[currency_pair].iloc[index]
+        if index != len(currency_pair_data[currency_pair])
+        else first_ask_bid(currency_pair, datetime.datetime.now())
+    )  # if not the end of the dataset we retrieve next dataset row
+
+    return [
+        html.P(
+            currency_pair, id=currency_pair, className="three-col"  # currency pair name
+        ),
+        html.P(
+            new_row[1].round(5),  # Bid value
+            id=new_row[0] + "bid",
+            className="three-col",
+            style={"color": get_color(new_row[1], bid)},
+        ),
+        html.P(
+            new_row[2].round(5),  # Ask value
+            className="three-col",
+            id=new_row[0] + "ask",
+            style={"color": get_color(new_row[2], ask)},
+        ),
+        html.Div(
+            index, id=currency_pair + "index", style={"display": "none"}
+        ),  # save index in hidden div
+    ]
+
+
+# Returns Top cell bar for header area
+def get_top_bar_cell(cellTitle, cellValue, color="white"):
+    return html.Div(
+        className="two-col",
+        children=[
+            html.P(className="p-top-bar", children=cellTitle),
+            html.P(id=cellTitle, children=cellValue),
+        ],
+    )
+
+
+# Returns HTML Top Bar for app layout
+def get_top_bar(
```

Comment:
When numbers get big and complicated, it gets hard to read. If it doesn't defeat the purpose of display these, might be better to trim the numbers and/or round them.

Reply:
![image](https://user-images.githubusercontent.com/26213135/60346179-9e2f1b00-9988-11e9-8ab3-704cadf13e61.png)

I feel like you lose a bunch of information when trimming the numbers. I feel like shrinking the text or something may help displaying the numbers work better.

Reply:
I added the compressed display format but only when the number exceeds 1M